packages/table/src/table.tsx

```diff
@@ -953,115 +953,119 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
-                    style.transform = `translate3d(0, ${-viewportRect.top}px, 0)`;
-                    return style;
-
-                default:
-                    return { display: "none" };
-            }
-        };
-        return this.maybeRenderRegions(styler);
+        return this.maybeRenderRegions(this.styleRowHeaderRegion);
+    }
+
+    private styleBodyRegion = (region: IRegion): React.CSSProperties => {
+        const cardinality = Regions.getRegionCardinality(region);
+        const style = this.grid.getRegionStyle(region);
+        switch (cardinality) {
+            case RegionCardinality.CELLS:
+                return style;
+
+            case RegionCardinality.FULL_COLUMNS:
+                style.top = "-1px";
+                return style;
+
+            case RegionCardinality.FULL_ROWS:
+                style.left = "-1px";
+                return style;
+
+            case RegionCardinality.FULL_TABLE:
+                style.left = "-1px";
+                style.top = "-1px";
+                return style;
+
+            default:
+                return { display: "none" };
+        }
+    }
+
+    private styleMenuRegion = (region: IRegion): React.CSSProperties => {
+        const { grid } = this;
+        const { viewportRect } = this.state;
+        if (viewportRect == null) {
+            return {};
+        }
+        const cardinality = Regions.getRegionCardinality(region);
+        const style = grid.getRegionStyle(region);
+
+        switch (cardinality) {
+            case RegionCardinality.FULL_TABLE:
+                style.right = "0px";
+                style.bottom = "0px";
+                style.top = "0px";
+                style.left = "0px";
+                style.borderBottom = "none";
+                style.borderRight = "none";
+                return style;
+
+            default:
+                return { display: "none" };
+        }
+    }
+
+    private styleColumnHeaderRegion = (region: IRegion): React.CSSProperties => {
+        const { grid } = this;
+        const { viewportRect } = this.state;
+        if (viewportRect == null) {
+            return {};
+        }
+        const cardinality = Regions.getRegionCardinality(region);
+        const style = grid.getRegionStyle(region);
+
+        switch (cardinality) {
+            case RegionCardinality.FULL_TABLE:
+                style.left = "-1px";
+                style.borderLeft = "none";
+                style.bottom = "-1px";
+                style.transform = `translate3d(${-viewportRect.left}px, 0, 0)`;
+                return style;
+            case RegionCardinality.FULL_COLUMNS:
+                style.bottom = "-1px";
+                style.transform = `translate3d(${-viewportRect.left}px, 0, 0)`;
+                return style;
+
+            default:
+                return { display: "none" };
+        }
+    }
+
+    private styleRowHeaderRegion = (region: IRegion): React.CSSProperties => {
```

Comment:
i wonder how much work it would be to make all these functions static? they could even live in a separate file.

you'd have to pull the `this.grid` and `this.state.viewportRect` logic out. maybe even accept an optional object of base styles to modify?